go/porcelain/deploy.go

```diff
@@ -384,6 +394,57 @@ func (n *Netlify) WaitUntilDeployLive(ctx context.Context, d *models.Deploy) (*m
 	return n.waitForState(ctx, d, "ready")
 }
 
+func (n *Netlify) uploadFilesWithSkipSameFileOptimization(ctx context.Context, d *models.Deploy, files *deployFiles, observer DeployObserver, t uploadType, timeout time.Duration) error {
+	sharedErr := &uploadError{err: nil, mutex: &sync.Mutex{}}
+	sem := make(chan int, n.uploadLimit)
+	wg := &sync.WaitGroup{}
+	defer wg.Wait()
+
+	var required []string
+	switch t {
+	case fileUpload:
+		required = d.Required
+	case functionUpload:
+		required = d.RequiredFunctions
+	}
+
+	count := 0
+	for _, sha := range required {
+		if files, exist := files.Hashed[sha]; exist {
+			for range files {
+				count++
+			}
+		}
+	}
+
+	log := context.GetLogger(ctx)
+	log.Infof("Uploading %v files", count)
```

Comment:
Will this count be greater than the files we actually upload? Since we upload the first file instead of all files that belong to a `sha`.

https://github.com/netlify/open-api/blob/fd8e4cb7f089086e99b9b4906d67fa5eb9f4c248/go/porcelain/deploy.go#L423-L425

Reply:
Yes it could be, this represents all the file refs we're going to upload. It just happens that some don't actually need it (and we log those accordingly).